### DISCUSSION TITLE
Control when turned on via MQTT command

### DISCUSSION THREAD

**Discussion by stephenmetzger** [basnijholt/adaptive-lighting#680] [Q&A]
Hi, 

My lights are zigbee bulbs, running in Zigbee2MQTT and are connected to sonoff mini devices that are controlled by standard US wall switches. Changes to the state of the wall switch (i.e., when the switch is flipped) triggers an MQTT publish of the payload "TOGGLE" to the Z2M topic (e.g. zigbee2mqtt/entry_light/set). There (to my knowledge) is never a light.turn_on service call fired, and when lights are turned on via a wall switch, they remain in their last color temp state. 

When I turn the light on and off via HomeAssistant, they adapt to the correct color temp. 

Is this expected behavior? Do I need to configure in a special way so that state changes as a result of MQTT commands are picked up my Adaptive Lighting?

**Comment by Kaot93** [comment]
Try not using the latest 1.18 but the 1.17 version of AL. 

I'm currently having the same problem with my shelly Dimmers. Maybe I'm going to make a bug report.

**Comment by basnijholt** [comment]
I’ll fix that it works with toggle. For now you can enable detect_non_ha_changes as a workaround. That is required if you turn on lights outside of Home Assistant. Currently it doesn’t listen to toggle and considers it as outside.

**Reply by Kaot93** [reply]
That was an unexpected fast reply. Thank you @basnijholt 🥇 !

**Comment by basnijholt** [comment]
I believe I have fixed the toggle issue in the latest main branch. Would it be possible to try it out?

**Reply by stephenmetzger** [reply]
Thanks. Unfortunately, adaptive lighting still isn't working the way I need it to for my use case. Commands from HA or executed from nodered using the HA node palette (light.turn_on commands) work beautifully. But, I mainly interact with my lights from switches, and my switches are set up to send mqtt commands to zigbee2mqtt-based light groups. 

It's almost like i need the opposite of take-over control/detect non-HA changes. I want adaptive lighting to recognize non-HA events, but to "intercept" them and ensure that the lights ramp up or down to the correct color/brightness.

**Reply by basnijholt** [reply]
I see. Unfortunately, that is just not possible to do because there is nothing to intercept.
